### ISSUE TITLE
Cannot get accuracy for vgg19 in tflite

### ISSUE THREAD

**Issue by venkat-kittu** [tensorflow/tensorflow#37412]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: NO
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu16.04
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**: pip3(20.02)
- **TensorFlow version (use command below)**: 2.1
- **Python version**: 3.6.9
- **Bazel version (if compiling from source)**:2.0.0
- **GCC/Compiler version (if compiling from source)**: 7.4.0
- **CUDA/cuDNN version**: 10.1
- **GPU model and memory**: 1060 GTX - 6GB
- **Exact command to reproduce**: 
bazel-bin/tensorflow/lite/tools/accuracy/ilsvrc/imagenet_accuracy_eval --model_file=../tensorflow_tests/models/vgg19.tflite --ground_truth_images_path=../val_imagenet/ --ground_truth_labels=ground_truth.txt --model_output_labels=./labels.txt --output_file_path=../vgg19_output_acc.txt --num_images=10

### Describe the problem
I have been running tflite accuracy script as defined [here](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/tools/accuracy/ilsvrc) and I am able to run successfully, but when I am running the script for resnet or mobilenet, I am getting proper accurate results from top1 to top10, but when I am trying same with vgg19, I am not getting the  accuracy instead I am getting all values as zero or very low as below

Top 1, Top 2, Top 3, Top 4, Top 5, Top 6, Top 7, Top 8, Top 9, Top 10
0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000
0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000
0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000
0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000
0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000
0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000
0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000, 0.000
0.000, 0.000, 0.000, 12.500, 12.500, 12.500, 12.500, 12.500, 12.500, 12.500
0.000, 0.000, 0.000, 11.111, 22.222, 22.222, 22.222, 22.222, 22.222, 22.222

I tried changing the model_labels text file, but it's not working.

do I am making any mistake with vgg19 or what is happening?
Thanks in advance

**Comment by freedomtan** [comment]
Where did you get your model? if it's converted from [SLIM](https://github.com/tensorflow/models/tree/master/research/slim) checkpoints, notes that those VGG models were converted from caffe models so that they have different preprocessing part and classes (1000 instead of 1001).

**Comment by venkat-kittu** [comment]
I have taken the model from Keras API

**Comment by freedomtan** [comment]
note that there are two problems.

1. preprocessing: keras supports 3 kinds of pre-processing: caffe, tf, and pytorch [1]. It seems Keras' VGG19 is from Caffe too, so it needs Caffe preprocessing [2]. Mobilenet uses tf preprocessing [3].

2. number of classes: you need an off-by-one labels file 

[1] https://github.com/keras-team/keras-applications/blob/master/keras_applications/imagenet_utils.py#L18-L62
[2] https://github.com/keras-team/keras-applications/blob/master/keras_applications/vgg19.py#L20
[3] https://github.com/keras-team/keras-applications/blob/master/keras_applications/mobilenet.py#L75-L84

**Comment by venkat-kittu** [comment]
@freedomtan thank you for your answer, but for calculating accuracy using tflite tool, do I need to make changes in the code of tflite accuracy tool for integrating Caffe preprocessing in it?

**Comment by freedomtan** [comment]
Relevant code is there [1]. I never tried it personally.

[1] https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/tools/evaluation/stages

**Comment by venkat-kittu** [comment]
@freedomtan Thank you, I will check it out once.

**Comment by gadagashwini** [comment]
@venkat-kittu, Any update for the @freedomtan's comment. Thanks!

**Comment by gadagashwini** [comment]
@venkat-kittu, Were you able to resolve this issue. Thanks!

**Comment by venkat-kittu** [comment]
Sorry for the delay, I was able to solve the issue by running prediction image by image and got accuarcy, buy using preprocessing in this [link](https://github.com/calebrob6/imagenet_validation/blob/master/1.%20Preprocess%20ImageNet%20validation%20set.ipynb)
